### PR TITLE
Add Idp-Initiated SSO

### PR DESCRIPTION
**Pull request by devkiran**
## What does this PR do?

This PR add support for Idp-Initiated SAML authentication.

**How it works**

- IdP POST `SAMLResponse` to `/api/auth/saml/callback`
- App redirect users to `/auth/saml-idp` with `code` in query params
- App use the new credentials provider `saml-idp` to fetch the user by exchanging the `code`
- `defaultRedirectUrl` should be changed to `/auth/saml-idp` while adding SAML connections

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by devkiran** [comment]
@deepakprabhakara Please take a look at the PR.

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 1]
```diff
@@ -201,6 +202,61 @@ if (isSAMLLoginEnabled) {
...
+        }
+
+        const { oauthController } = await jackson();
+
+        // Fetch access token
+        const { access_token } = await oauthController.token({
+          code,
+          grant_type: "authorization_code",
+          redirect_uri: `${process.env.NEXTAUTH_URL}`,
+          client_id: "dummy",
```
Comment (emrysal) [inline apps/web/pages/api/auth/[...nextauth].tsx:232]: Should this be configurable or not be dummy?

Comment (devkiran) [reply apps/web/pages/api/auth/[...nextauth].tsx:232]: We can leave it as `dummy` or configure `CLIENT_SECRET_VERIFIER`. [More info](https://boxyhq.com/docs/jackson/deploy/env-variables#client_secret_verifier)

Comment (alannnc) [reply apps/web/pages/api/auth/[...nextauth].tsx:232]: We should go for CLIENT_SECRET_VERIFIER in .env, if that doesn't exists then switch to default dummy.

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -280,7 +336,7 @@ export default NextAuth({
         return await autoMergeIdentities();
       }
 
-      if (account && account.type === "credentials") {
+      if (account && account.type === "credentials" && account.provider != "saml-idp") {
```
Comment (alannnc) [inline apps/web/pages/api/auth/[...nextauth].tsx:339]:
```suggestion
      if (account && account.type === "credentials" && account.provider !== "saml-idp") {
```

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 3]
```diff
@@ -385,18 +446,18 @@ export default NextAuth({
           return "/auth/error?error=unverified-email";
         }
         // Only google oauth on this path
-        const provider = account.provider.toUpperCase() as IdentityProvider;
+        // const provider = account.provider.toUpperCase() as IdentityProvider;
```
Comment (alannnc) [inline apps/web/pages/api/auth/[...nextauth].tsx:449]:
```suggestion
```
Remove if no longer required

**Review by zomars — APPROVED** [review]

**Comment by zomars** [comment]
Thank you for your contribution 🙏 Just waiting for tests to pass

**Comment by zomars** [comment]
Manually merging to bypass unrelated failing e2e test.

**Pull request merged**